### PR TITLE
feat: publish gifs

### PR DESCRIPTION
**Pull request by aymanbagabas**
Publish GIF images to our `vhs.charm.sh` hosting.

* Add `publish` command
* Add `--publish` flag

### CODE REVIEW

File: publish.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+	signer, err := ssh.NewSignerFromKey(kp.PrivateKey())
+	if err != nil {
+		return nil, err
+	}
+
+	pkam := ssh.PublicKeys(signer)
+	sshConfig := &ssh.ClientConfig{
+		User:            "vhs",
+		Auth:            []ssh.AuthMethod{pkam},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(), //nolint
```
Comment (maaslalani) [inline publish.go:65]:
```suggestion
		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
```

File: main.go [hunk 2]
```diff
@@ -163,14 +179,17 @@ func main() {
 }
 
 func init() {
+	rootCmd.Flags().BoolVarP(&publish, "publish", "p", false, "publish the GIF")
```
Comment (maaslalani) [inline main.go:182]:
```suggestion
	rootCmd.Flags().BoolVarP(&publish, "publish", "p", false, "Publish your GIF to vhs.charm.sh and get a shareable URL")
```

File: publish.go [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+)
+
+const (
+	ghostHost = "ghost.charm.sh"
+	ghostPort = 22
+)
+
+var publishCmd = &cobra.Command{
+	Use:           "publish <gif>",
+	Short:         "Publish a GIF to the web",
```
Comment (maaslalani) [inline publish.go:23]:
```suggestion
	Short:         "Publish your GIF to vhs.charm.sh and get a shareable URL",
```

**Review by maaslalani — APPROVED** [review]
This is so cool! Amazing work, Ayman! Thank you so much! Feel free to merge whenever you're ready!

File: publish.go (hunk 1 above)
Comment (caarlos0) [reply publish.go:65]: could probably be copied from: https://github.com/charmbracelet/wishlist/blob/43ce3d6a27c6e54a880b643d4de9f8ac7f578a85/client_auth.go#L254

**Pull request merged**